### ISSUE TITLE
Release CAS 4.1.0 RC1

### ISSUE THREAD

**Issue by mmoayyed** [Jasig/cas#1097]
Waiting for #1093 to merge.

**Comment by leleuj** [comment]
@mmoayyed : will start the RC1 release in a few minutes unless someone has a concern.

**Comment by mmoayyed** [comment]
Go for it. I'll be online in case you need a hand.

**Comment by leleuj** [comment]
Thanks. I've checked issues, I'm taking a look at why latest builds have failed in Travis.

**Comment by leleuj** [comment]
Fixed some NOTICEs, updated the version to 4.1.0-RC1-SNAPSHOT.

**Comment by leleuj** [comment]
@mmoayyed : I encountered a few issues. I'm still in the process of the release. I won't be able to finish the release until Monday. Please don't commit anything.

**Comment by mmoayyed** [comment]
Copy that.

**Comment by mmoayyed** [comment]
@leleuj were you planning to get back to this on Monday? or can we expect something during the weekend?

**Comment by leleuj** [comment]
Unfortunately, I use Windows at home and my environment is not properly set up for releasing. I'll continue the release at work with my Mac on Monday.

**Comment by mmoayyed** [comment]
You probably want to reset the master back to this commit:
https://github.com/Jasig/cas/commit/22bb4100c6849685fcfe110dca923afdaf76ba61

Right now there are a few commits in the master branch that are produced by the maven release plugin. Those should be removed when you restart.

**Comment by leleuj** [comment]
A little update as the "nightmare" goes on.

I started: `mvn release:perform`, but it ended up in error with:

```
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.7:jar (attach-javadocs) on project cas-server-core: MavenReportException: Error while creating archive:
    [ERROR] Exit code: 1 - com.sun.tools.javac.code.Symbol$CompletionFailure: class file for net.sourceforge.cobertura.coveragedata.LightClassmapListener not found
    [ERROR] javadoc: error - com.sun.tools.javac.code.Symbol$CompletionFailure: class file for net.sourceforge.cobertura.coveragedata.LightClassmapListener not found
    [ERROR]
    [ERROR] Command line was: /Library/Java/JavaVirtualMachines/jdk1.8.0_40.jdk/Contents/Home/bin/javadoc @options @packages
```

It seems related to the JDK 8. I'm restarting the `release:perform` stage using the JDK 7, the build is hanging on the `jdepend-maven-plugin` for now...

**Comment by mmoayyed** [comment]
You should definitely do prepare and release actions under JDK7. Also, run both with the nocheck profile.

**Comment by battags** [comment]
It may be time to try a different code coverage tool. :-) [not for this release]

**Comment by leleuj** [comment]
It should work with the JDK 8 as well. The `nocheck` profile is already used by the `maven-release-plugin`: https://github.com/Jasig/cas/blob/master/pom.xml#L490. I don't need to call it explicitly.
But indeed, I will try to disable the cobertura plugin for the release.

**Comment by leleuj** [comment]
And with the JDK 7, I just got the error: `Exception in thread "main" java.lang.OutOfMemoryError: PermGen space`... Error I would not have had using the JDK 8 which has no PermGen Space...

I'll update this doc: http://jasig.github.io/cas/developer/Release-Process.html at the end.

**Comment by mmoayyed** [comment]
Have you set `export MAVEN_OPTS="-Xmx2048m"` or something equivalent?

**Comment by leleuj** [comment]
Yes, but `-Xmx2048m` is for the Heap memory. I just used: `-XX:MaxPermSize=128m` for PermGen space... In progress...

**Comment by leleuj** [comment]
The release has been cut. It should be available in the Maven central repository in a few hours. I'll make the announcement as well as the doc updates on Wednesday. Unless @mmoayyed wants to make the announcement earlier.

**Comment by mmoayyed** [comment]
Perfect, thank you. I think we learn a whole new bunch of stuff every time we cut a release :)

Go right ahead; I'll make the docs available for 4.1.
